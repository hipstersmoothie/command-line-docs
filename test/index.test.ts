import { Command, MultiCommand } from 'command-line-application';

import commandLineDocs from '../src';

describe('single command app', () => {
  test('basic', () => {
    const echo: Command = {
      name: 'echo',
      description: 'Print a string to the terminal',
      options: [
        {
          name: 'value',
          type: String,
          defaultOption: true,
          description: 'The value to print',
        },
      ],
    };

    expect(commandLineDocs(echo)).toMatchSnapshot();
  });

  test('with examples', () => {
    const echo: Command = {
      name: 'echo',
      description: 'Print a string to the terminal',
      examples: ['echo foo', 'echo "Intense message"'],
      options: [
        {
          name: 'value',
          type: String,
          defaultOption: true,
          description: 'The value to print',
        },
      ],
    };

    expect(commandLineDocs(echo)).toMatchSnapshot();
  });
});

test('MultiCommand', () => {
  const testCommand: Command = {
    name: 'test',
    description: 'test the project',
    examples: ['test --interactive'],
    options: [
      {
        name: 'interactive',
        type: Boolean,
        description: 'Run the application in interactive mode',
      },
    ],
  };
  const lintCommand: Command = {
    name: 'lint',
    description: 'lint the project',
    examples: ['lint --fix'],
    options: [
      {
        name: 'fix',
        type: Boolean,
        description: 'Run the application in fix mode',
      },
    ],
  };
  const scripts: MultiCommand = {
    name: 'scripts',
    description: 'My scripts package',
    commands: [lintCommand, testCommand],
    options: [
      {
        name: 'verbose',
        alias: 'v',
        description: 'Log a bunch of stuff',
        type: Boolean,
      },
    ],
  };

  expect(commandLineDocs(scripts)).toMatchSnapshot();
});
