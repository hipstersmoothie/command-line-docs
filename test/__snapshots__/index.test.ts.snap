// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiCommand 1`] = `
"# \`scripts\`

My scripts package

## Commands

  - **lint** - lint the project
  - **test** - test the project

## Options

| Flag | Type | Description |
| - | - | - |
| \`--verbose\`, \`-v\` | Boolean | Log a bunch of stuff |

"
`;

exports[`single command app 1`] = `
"# \`echo\`

Print a string to the terminal

## Options

| Flag | Type | Description |
| - | - | - |
| \`--value\` | String | The value to print |

"
`;
